osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/ServiceFunctionChainCheckMetaTask.java

```diff
@@ -0,0 +1,141 @@
+public class ServiceFunctionChainCheckMetaTask extends TransactionalMetaTask {
+	@Reference
+	CreateServiceFunctionChainTask createServiceFunctionChainTask;
+
+	@Reference
+	UpdateServiceFunctionChainTask updateServiceFunctionChainTask;
+
+	@Reference
+	DeleteServiceFunctionChainTask deleteServiceFunctionChainTask;
+
+	@Reference
+	private ApiFactoryService apiFactoryService;
+
+	private SecurityGroup securityGroup;
+	TaskGraph tg;
+	private List<NetworkElement> portPairGroups;
+
+	public ServiceFunctionChainCheckMetaTask create(SecurityGroup sg) {
+		ServiceFunctionChainCheckMetaTask task = new ServiceFunctionChainCheckMetaTask();
+
+		task.createServiceFunctionChainTask = this.createServiceFunctionChainTask;
+		task.securityGroup = sg;
+		task.dbConnectionManager = this.dbConnectionManager;
+		task.txBroadcastUtil = this.txBroadcastUtil;
+
+		return task;
+	}
+
+	@Override
+	public void executeTransaction(EntityManager em) throws Exception {
+		this.tg = new TaskGraph();
+		this.securityGroup = em.find(SecurityGroup.class, this.securityGroup.getId());
+
+		if (this.securityGroup.getNetworkElementId() == null && this.securityGroup.getServiceFunctionChain() != null) {
+			this.tg.appendTask(this.createServiceFunctionChainTask.create(this.securityGroup.getServiceFunctionChain(),
+					this.securityGroup));
+		} else {
+			if (SecurityGroupEntityMgr.listOtherSecurityGroupsWithSameSFC(em, this.securityGroup).size() < 1) {
+
+				this.tg.appendTask(this.deleteServiceFunctionChainTask
+						.create(this.securityGroup.getServiceFunctionChain(), this.securityGroup));
+
+			} else {
+				this.portPairGroups = getPortPairGroupWithSameProjectId(em, this.securityGroup.getProjectId(),
+						this.securityGroup.getServiceFunctionChain().getVirtualSystems());
+
+				try (SdnRedirectionApi sdnApi = this.apiFactoryService
+						.createNetworkRedirectionApi(this.securityGroup.getVirtualizationConnector())) {
+
+					List<NetworkElement> sfcPortPairGroups = sdnApi
+							.getNetworkElements(new NetworkElementImpl(this.securityGroup.getNetworkElementId()));
+
+					if (!this.portPairGroups.equals(sfcPortPairGroups)) {
+						this.tg.appendTask(this.updateServiceFunctionChainTask.create(
+								this.securityGroup.getServiceFunctionChain(), this.securityGroup, this.portPairGroups));
+					}
+				}
+			}
+		}
+	}
+
+	static List<NetworkElement> getPortPairGroupWithSameProjectId(EntityManager em, String projectId,
+			List<VirtualSystem> vsList) throws VmidcBrokerValidationException {
+
+		List<NetworkElement> portPairGroups = new ArrayList<>();
+
+		for (VirtualSystem vs : vsList) {
+			if (vs.getDeploymentSpecs() == null || vs.getDeploymentSpecs().isEmpty()) {
+				throw new VmidcBrokerValidationException(
+						String.format("No deployment found for virtual system %s.", vs.getName()));
+			}
+
+			if (DeploymentSpecEntityMgr.findDeploymentSpecsByVirtualSystemProject(em, vs, projectId).size() > 1) {
```

Comment:
add a todo here. DS is unique across vs,project AND region. SG is not tied any region so we have to handle some edge cases here. Please think about the cases and add them to the comments